contracts/interchain-token-service/src/contract.rs

```diff
@@ -186,6 +188,60 @@ impl InterchainTokenServiceInterface for InterchainTokenService {
             .into()
     }
 
+    fn interchain_token_id(env: &Env, sender: Address, salt: BytesN<32>) -> BytesN<32> {
+        env.crypto()
+            .keccak256(&(PREFIX_INTERCHAIN_TOKEN_ID, sender, salt).to_xdr(env))
+            .into()
+    }
+
+    fn deploy_interchain_token(
+        env: &Env,
+        caller: Address,
+        salt: BytesN<32>,
+        token_meta_data: TokenMetadata,
+        initial_supply: i128,
+        _minter: Address,
+    ) -> Result<BytesN<32>, ContractError> {
+        caller.require_auth();
+
+        let minter = if initial_supply > 0 {
+            env.current_contract_address()
+        } else {
+            ensure!(
+                _minter != env.current_contract_address(),
+                ContractError::InvalidMinter
+            );
+            _minter
+        };
+
+        let deploy_salt = Self::interchain_token_deploy_salt(env, caller.clone(), salt.clone());
+        let interchain_token_wasm_hash = Self::interchain_token_wasm_hash(env);
+        let token_id = Self::interchain_token_id(env, caller.clone(), salt);
+
+        let deployed_address = env
+            .deployer()
+            .with_address(caller.clone(), deploy_salt)
+            .deploy_v2(
+                interchain_token_wasm_hash,
```

Comment:
```suggestion
                Self::interchain_token_wasm_hash(env),
```
local variable isn't needed